src/mame/tecmo/spbactn.cpp

```diff
@@ -481,15 +496,24 @@ void spbactn_state::sound_map(address_map &map)
 }
 
 
+uint8_t spbactnp_state::extra_latch_r(offs_t offset)
+{
+	uint16_t latch = m_extralatch->read();
+	if (offset & 1) latch >>= 8;
```

Comment:
This needs a check for suppressed side effects or it will break in the debugger.

Reply:
I don't see how.
read() in the gen latch device checks for it itself (otherwise it'd show the 'read before write' warning in the debugger)

Reply:
Gah, I’m too tired, I thought it was changing a member.  It’s be more obvious it isn’t trying to change state if it was like `return (offset & 1) ? (latch >> 8) : (latch & 0xff);`

Reply:
I don't see how.
read() in the gen latch device checks for it itself (otherwise it'd show the 'read before write' warning in the debugger)

Reply:
I didn’t realise `latch` was a local when I first read it – I thought it was shifting the latch by eight bits on reads at offset 1.  If it was written in a way that didn’t require modifying the local (more “functional programming”-esque) it wouldn’t be possible to misread it that way.